File: 1-easy/problems/3142-check-if-grid-satisfies-conditions/solution.ts
function satisfiesConditions(grid: any): boolean | number | string | any {
  """
  :type grid: List[List[int]]
  :rtype: bool
  """
  return (all(grid[i][j] == grid[i+1][j] for j in Array.from({length: (grid[0]}, (_, i) => i).length) for i in Array.from({length: (grid}, (_, i) => i).length-1)) && ;
          all(grid[0][j] != grid[0][j+1] for j in Array.from({length: (grid[0]}, (_, i) => i).length-1)))
}

export { satisfiesConditions };

File: 1-easy/problems/3142-check-if-grid-satisfies-conditions/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def satisfiesConditions(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: bool
        """
        return (all(grid[i][j] == grid[i+1][j] for j in range(len(grid[0])) for i in range(len(grid)-1)) and 
                all(grid[0][j] != grid[0][j+1] for j in range(len(grid[0])-1)))
